File: js/트리의 부모 찾기(11725).js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const N = Number(input[0]);

const graph = Array.from({ length: N + 1 }, () => []);
const visited = Array(N + 1).fill(false);
const parent = Array(N + 1).fill(0);

for (let i = 1; i < N; i++) {
    const [s, e] = input[i].split(' ').map(Number);
    graph[s].push(e);
    graph[e].push(s);
}

function dfs(v) {
    visited[v] = true;
    for (let i of graph[v]) {
        if (!visited[i]) {
            parent[i] = v;
            dfs(i);
        }
    }
}

dfs(1);

// ✅ `parent[2]`부터 출력해야 함
console.log(parent.slice(2).join("\n"));

File: python/트리의 부모 찾기(11725).py
import sys
sys.setrecursionlimit(10**6)
input = sys.stdin.readline

n = int(input())
visited = [0] * (n+1)
parent = [0] * (n+1)
graph = [[] for _ in range(n+1)]

for _ in range(n-1):
    s, e = map(int, input().split())
    graph[s].append(e)
    graph[e].append(s)

def dfs(v):
    visited[v] = 1
    for i in graph[v]:
        if not visited[i]:
            parent[i] = v
            dfs(i)

dfs(1)
print(*parent[2:], sep='\n')